package/src/index.ts

```diff
@@ -0,0 +1,537 @@
+	 */
+    constructor(private token?: string) {}
+
+    /**
+	 * Creates {@link default.rankImages} and {@link default.mapImages} by pulling the images from {@link https://valorant-api.com}
+	 * @remarks
+	 * Must be called before using {@link default.rankImages} and {@link default.mapImages}
+	 */
+    async initUtils(): Promise<void> {
+        const mapsReq = (await axios({ url: "https://valorant-api.com/v1/maps" }).catch(() => null))?.data?.data
+        if (mapsReq) {
+            this.mapImages = {} as any
+            mapsReq.forEach((map: any) => (this.mapImages[map.displayName] = {
+                splash: map.splash,
+                minimap: map.displayIcon,
+                landscape: map.listViewIcon
+            }))
+        }
+
+        const tiersReq = (await axios({ url: "https://valorant-api.com/v1/competitivetiers" }).catch(() => null))?.data?.data
+        if (tiersReq) {
+            const capitalize = (string: string) => string.charAt(0).toUpperCase() + string.slice(1)
+            this.rankImages = {} as any
+            tiersReq[tiersReq.length - 1].tiers.forEach((rank: any) => (this.rankImages[capitalize(rank.tierName.toLowerCase())] = {
+                large: rank.largeIcon,
+                small: rank.smallIcon,
+                triangleUp: rank.rankTriangleUpIcon,
+                triangleDown: rank.rankTriangleDownIcon
+            }))
+        }
+    }
+
+    /**
+	 * @internal
+	 * Parses the body of an axios response
+	 */
+    private _parseBody(body: any): any {
+        if (body.errors) return body.errors
+        return body.status ? body.data : body
+    }
+
+    /**
+	 * @internal
+	 * Main function to fetch from the API
+	 * @typeParam dataType - The type of the response
+	 * @param url - URl to fetch. Appended to the base url and automatically encoded
+	 * @param searchParams - Any extra search params. Empty values will be filtered out
+	 * @param config - Override the default axios config with custom params
+	 * @returns Formatted response
+	 */
+    private async _fetch<dataType>(url: string, searchParams?: { [key: string]: any }, config: AxiosRequestConfig = {}): Promise<APIResponse<dataType>> {
+        // Format search params into `?something=like&this`
+        const queryParams = searchParams ? Object.keys(searchParams).reduce((acc, cur) => {
+            const value = searchParams[cur]
+            if (value === null || value === undefined) return acc
+            return acc === ""
+                ? `?${cur}=${value}`
+                : acc + `&${cur}=${value}`
+        }, "") : ""
+
+        // Main request
+        const req = await axios({
+            url: encodeURI(`${baseUrl}/${url}${queryParams}`),
+            responseType: "json",
+            headers: this.token
+                ? { Authorization: this.token, "User-Agent": "unofficial-valorant-api/node.js/2.3.0", }
+                : { "User-Agent": "unofficial-valorant-api/node.js/2.3.0", },
```

Comment:
Why adding the version number here? Needs to be updated whenever a new version gets published?

Reply:
This was taken from the original `index.js`. But I'll read it from the `package.json`